group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java

```diff
@@ -146,14 +153,16 @@ GroupMetadataManager build() {
             if (metadataImage == null) metadataImage = MetadataImage.EMPTY;
             if (time == null) time = Time.SYSTEM;
 
-            if (assignors == null || assignors.isEmpty()) {
+            if (timer == null)
+                throw new IllegalArgumentException("Timer must be set.");
```

Comment:
Should we throw illegal argument here but illegal state for the assignors?

Reply:
Good catch. I think that we should use `IllegalArgumentException` for the assignors as well.